e2e/scripts/generate-controller-join-token.js

```diff
@@ -0,0 +1,272 @@
+            targetPage.locator('button'),
+            targetPage.locator('::-p-xpath(//*[@id=\\"app\\"]/div/div[2]/div/div/div/div[2]/div/div/div[2]/button)'),
+            targetPage.locator(':scope >>> button')
+        ])
+            .setTimeout(timeout)
+            .click({
+              offset: {
+                x: 30,
+                y: 14.0078125,
+              },
+            });
+    }
+    {
+        process.stderr.write("waiting and clicking in the Cluster Management tab\n");
+        const targetPage = page;
+        await puppeteer.Locator.race([
+            targetPage.locator('div:nth-of-type(3) > span'),
+            targetPage.locator('::-p-xpath(//*[@id=\\"app\\"]/div/div[1]/div[1]/div[2]/div[3]/span)'),
+            targetPage.locator(':scope >>> div:nth-of-type(3) > span'),
+            targetPage.locator('::-p-text(Cluster Management)')
+        ])
+            .setTimeout(timeout)
+            .click({
+              offset: {
+                x: 87.734375,
+                y: 26,
+              },
+            });
+    }
+    {
+        process.stderr.write("waiting and clicking in the Add node button\n");
+        const targetPage = page;
+        await puppeteer.Locator.race([
+            targetPage.locator('::-p-aria(Add node)'),
+            targetPage.locator('div.tw-flex > button'),
+            targetPage.locator('::-p-xpath(//*[@id=\\"app\\"]/div/div[2]/div/div/div[1]/button)'),
+            targetPage.locator(':scope >>> div.tw-flex > button'),
+            targetPage.locator('::-p-text(Add node)')
+        ])
+            .setTimeout(timeout)
+            .click({
+              offset: {
+                x: 24.328125,
+                y: 17,
+              },
+            });
+    }
+    {
+        process.stderr.write("waiting and clicking in the controller role\n");
+        const targetPage = page;
+        await puppeteer.Locator.race([
+            targetPage.locator('div:nth-of-type(1) > label'),
+            targetPage.locator('::-p-xpath(/html/body/div[5]/div/div/div/div[2]/div[1]/label)'),
+            targetPage.locator(':scope >>> div:nth-of-type(1) > label')
+        ])
+            .setTimeout(timeout)
+            .click({
+              offset: {
+                x: 110,
+                y: 27.5,
+              },
+            });
+    }
+    {
+        // CUSTOM: finding the element that contains the node join command.
+        process.stderr.write("waiting and fetching the node join command\n");
+        const targetPage = page;
+        await targetPage.waitForSelector('.react-prism.language-bash');
+        let elementContent = await targetPage.evaluate(() => {
+            const element = document.querySelector('.react-prism.language-bash');
+            return element ? element.textContent : null;
+        });
+        if (!elementContent) {
+            throw new Error("Could not find the node join command");
+        }
+        console.log(elementContent);
```

Comment:
Could we log a json object instead to simplify error logging in the future?

Reply:
Done. While doing it I also added a script to generate a worker token as well.